internal/proxy/reverse_proxy_test.go

```diff
@@ -484,6 +484,16 @@ func TestTimeoutHandler(t *testing.T) {
 			ExpectedStatusCode: 200,
 			ExpectedBody:       "OK",
 		},
+		{
+			name: "does not time out as flush interval is set",
+			config: &UpstreamConfig{
+				Timeout:       time.Duration(100) * time.Millisecond,
```

Comment:
Taking a deep look at this PR there are some issues with these tests:

1. We want to set a duration below `50` since that's how long we need to wait for the server below to throw an error. Any higher and we wouldn't timeout anyway.

2. This test doesn't construct an upstream reverse proxy! This set of tests just verifies the timeout handler behavior. To fix,  we could set that up here and add some more setup below or we need to move this to a set of tests where the the upstream reverse proxy is setup and used.

Reply:
Great catch - I was a little bit too hasty in getting this added in!